NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/SrpClient.cs

```diff
@@ -0,0 +1,239 @@
+			BigInteger vu = BigInteger.ModPow(v, u, N);
+			BigInteger Avu;
+			BigInteger.DivRem(BigInteger.Multiply(A, vu), N, out Avu);
+			BigInteger sessionSecret = BigInteger.ModPow(Avu, b, N);
+			return sha1(toBigByteArray(sessionSecret));
+		}
+
+		public SrpClient() {
+			privateKey = getSecret();
+			publicKey = BigInteger.ModPow(g, privateKey, N);
+		}
+
+		public BigInteger getPublicKey() {
+			return publicKey;
+		}
+
+		public BigInteger getPrivateKey() {
+			return privateKey;
+		}
+
+		private byte[] getClientSessionKey(String user, String password, byte[] salt, BigInteger serverPublicKey) {
+			BigInteger u = getScramble(publicKey, serverPublicKey);
+			BigInteger x = getUserHash(user, password, salt);
+			BigInteger gx = BigInteger.ModPow(g, x, N);
+			BigInteger kgx;
+			BigInteger.DivRem(BigInteger.Multiply(k, gx), N, out kgx);
+			BigInteger Bkgx = BigInteger.Subtract(serverPublicKey, kgx);	// B-kgx
+			if (BigInteger.Compare(Bkgx, 0) < 0) {
+				Bkgx = BigInteger.Add(Bkgx, N);
+			}
+			BigInteger diff;
+			BigInteger.DivRem(Bkgx, N, out diff);
+			BigInteger ux;
+			BigInteger.DivRem(BigInteger.Multiply(u, x), N, out ux);
+			BigInteger aux;
+			BigInteger.DivRem(BigInteger.Add(privateKey, ux), N, out aux);
+			BigInteger sessionSecret = BigInteger.ModPow(diff, aux, N);
+
+			return sha1(toBigByteArray(sessionSecret));
+		}
+
+		public String getPublicKeyHex() {
+			return toHexString(pad(publicKey));
+		}
+
+		public byte[] clientProof(String user, String password, byte[] salt, BigInteger serverPublicKey) {
+			byte[] K = getClientSessionKey(user, password, salt, serverPublicKey);
+
+			BigInteger n1 = fromBigByteArray(sha1(toBigByteArray(N)));
+			BigInteger n2 = fromBigByteArray(sha1(toBigByteArray(g)));
+
+			n1 = BigInteger.ModPow(n1, n2, N);
+			n2 = fromBigByteArray(sha1(System.Text.Encoding.UTF8.GetBytes(user.ToUpper())));
+			byte[] M = sha1(toBigByteArray(n1), toBigByteArray(n2), salt, toBigByteArray(publicKey), toBigByteArray(serverPublicKey), K);
+
+			sessionKey = K;
+			proof = M;
+
+			return proof;
+		}
+
+		public byte[] clientProof(String user, String password, byte[] authData) {
+			int saltLength = authData[0] + authData[1] * 256;
+			byte[] salt = new byte[saltLength];
+			Array.Copy(authData, 2, salt, 0, saltLength);
+
+			int serverKeyStart = saltLength + 4;
+			int serverKeyLength = authData.Length - saltLength - 4;
+			byte[] hexServerPublicKey = new byte[serverKeyLength];
+			Array.Copy(authData, serverKeyStart, hexServerPublicKey, 0, serverKeyLength);
+			String hexServerPublicKeyString = System.Text.Encoding.UTF8.GetString(hexServerPublicKey);
+			BigInteger serverPublicKey = BigInteger.Parse("00" + hexServerPublicKeyString, NumberStyles.HexNumber);
+			return clientProof(user.ToUpper(), password, salt, serverPublicKey);
+		}
+
+		public byte[] getSessionKey() {
+			return sessionKey;
+		}
+
+		public static void Main(string[] args) {
```

Comment:
Main() ?
It's my test code to verify srp authentication.
Sorry It my working test code, so It can remove.